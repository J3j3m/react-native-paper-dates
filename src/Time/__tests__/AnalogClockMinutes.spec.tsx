import React from 'react'
import { render } from '@testing-library/react-native'
import AnalogClockMinutes, { getMinuteNumbers } from '../AnalogClockMinutes'
import { circleSize } from '../timeUtils'

describe('AmPmSwitcher', () => {
  it('should return minutes numbers coordinates', () => {
    // TODO need to be challenged here
    const numbers = getMinuteNumbers(circleSize, 12)
    expect(numbers).toEqual([
      [108, 22],
      [151, 33],
      [182, 65],
      [194, 107],
      [182, 150],
      [151, 182],
      [108, 194],
      [65, 182],
      [33, 151],
      [22, 108],
      [33, 65],
      [64, 33],
    ])
  })
  it('should render analog clock', () => {
    const { toJSON } = render(<AnalogClockMinutes {...{ minutes: 24 }} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
