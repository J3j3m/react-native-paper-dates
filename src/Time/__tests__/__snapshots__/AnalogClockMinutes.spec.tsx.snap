// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmPmSwitcher should render analog clock 1`] = `
Array [
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 108,
          "top": 22,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        00
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 151,
          "top": 33,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        5
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 182,
          "top": 65,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        10
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 194,
          "top": 107,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        15
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 182,
          "top": 150,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        20
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 151,
          "top": 182,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        25
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 108,
          "top": 194,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        30
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 65,
          "top": 182,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        35
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 33,
          "top": 151,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        40
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 22,
          "top": 108,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        45
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 33,
          "top": 65,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        50
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginLeft": -25,
          "marginTop": -25,
          "position": "absolute",
          "width": 50,
          "zIndex": 20,
        },
        Object {
          "left": 64,
          "top": 33,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
        }
      }
    >
      <Text
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            null,
          ]
        }
      >
        55
      </Text>
    </View>
  </View>,
]
`;
